solutions/usaco-690.mdx

```diff
@@ -69,4 +69,34 @@ int main () {
 
 </CPPSection>
 
+<PySection>
+
+```python
+import heapq
+
+with open('cowdance.in') as r:
+	n, maximum = map(int, r.readline().split())
+	dance = [int(r.readline()) for _ in range(n)]
+
+left = 1
+right = n + 1
+while left < right:
+	mid = (left + right) // 2
+	parts = dance[:mid]
+	heapq.heapify(parts)
+
+	for i in range(mid, n): 
+		tmp = heapq.heappop(parts) + dance[i]
```

Comment:
a name better than just `tmp` would be nice

Reply:
okay

Reply:
recommendation: exit_stage_time

Reply:
recommendation: `exit_stage_time`

Reply:
okay